src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -189,6 +189,402 @@ public static class ILoggerExtensions
+        public static void LogServiceBusQueueRequestWithPrefix(
+            this ILogger logger,
+            string serviceBusNamespacePrefix,
+            string queueName,
+            string operationName,
+            bool isSuccessful,
+            TimeSpan duration,
+            DateTimeOffset startTime,
+            Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger), "Requires an logger instance to track telemetry");
+            Guard.NotNullOrWhitespace(serviceBusNamespacePrefix, nameof(serviceBusNamespacePrefix), "Requires an Azure Service Bus namespace to track the queue request");
+            Guard.NotNullOrWhitespace(queueName, nameof(queueName), "Requires an Azure Service Bus queue name to track the queue request");
+            Guard.NotLessThan(duration, TimeSpan.Zero, nameof(duration), "Requires a positive time duration of the Azure Service Bus queue request operation");
+
+            LogServiceBusRequestWithPrefix(logger, serviceBusNamespacePrefix, queueName, operationName, isSuccessful, duration, startTime, ServiceBusEntityType.Queue, context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Service Bus queue request.
+        /// </summary>
+        /// <param name="logger">The logger instance to track the telemetry.</param>
+        /// <param name="serviceBusNamespace">The namespace where the Azure Service Bus queue is registered.</param>
+        /// <param name="queueName">The name of the Azure Service Bus queue.</param>
+        /// <param name="operationName">The name of the operation of the request.</param>
+        /// <param name="isSuccessful">The indication whether or not the Azure Service Bus queue request was successfully processed.</param>
+        /// <param name="duration">The duration it took to process the Azure Service Bus queue request.</param>
+        /// <param name="startTime">The time when the request was received.</param>
+        /// <param name="context">The telemetry context that provides more insights on the Azure Service Bus queue request.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="serviceBusNamespace"/> or <paramref name="queueName"/> is blank.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="duration"/> is a negative time range.</exception>
+        public static void LogServiceBusQueueRequest(
+            this ILogger logger,
+            string serviceBusNamespace,
+            string queueName,
+            string operationName,
+            bool isSuccessful,
+            TimeSpan duration,
+            DateTimeOffset startTime,
+            Dictionary<string, object> context = null)
```

Comment:
Just asking: do we need to make a distinction between topics and queues here ?  

If so, then I'd suggest that we extend the 'TopicRequest' method signatures with a `subscriptionName` parameter, as you receive messages that are posted on a topic through a subscription.

Reply:
We do a distinction on the `ServiceBusEntityType`, yeah. Ah, indeed. An extra parameter for this would be nice. Thx!

Reply:
Added topic subscription in https://github.com/arcus-azure/arcus.observability/pull/250/commits/361b9d0c65a3c438165e78a5030282fa757dea40